### ISSUE TITLE
Enable Mermaid on Jan and Nitro Docs

### ISSUE THREAD

**Issue by dan-jan** [janhq/jan#654]
## Objective

Jan and Nitro docs should render Mermaid diagrams. @vuonghoainam and I will be creating architecture diagrams using Mermaid. 

## Bugs

I tried to enable Mermaid by following the instructions in Docusaurus Mermaid section. 

https://docusaurus.io/docs/markdown-features/diagrams

https://docusaurus.io/docs/api/themes/@docusaurus/theme-mermaid

However, I encounter the following error: 

```
Module not found: Error: Can't resolve '@docusaurus/useIsomorphicLayoutEffect' in '/Users/daniel/jan/jan/docs/node_modules/@docusaurus/theme-mermaid/node_modules/@docusaurus/theme-common/lib/components/Collapsible'
Module not found: Error: Can't resolve '@docusaurus/useIsomorphicLayoutEffect' in '/Users/daniel/jan/jan/docs/node_modules/@docusaurus/theme-mermaid/node_modules/@docusaurus/theme-common/lib/utils'
client (webpack 5.89.0) compiled with 2 errors and 2 warnings
```

**Comment by dan-jan** [comment]
cc @vuonghoainam FYI, there is a bug in Mermaid <> Docusaurus, will get @hahuyhoang411 @hieu-jan to take a closer look

**Comment by hahuyhoang411** [comment]
Will be back with this later. Spent an hour to fix but did not work.

**Comment by dan-jan** [comment]
Maybe upgrading Docusaurus will fix it? We seem to be at 2.4, while Mermaid's native install is at 3

**Comment by hieu-jan** [comment]
I completed enabling Mermaid

https://github.com/janhq/jan/pull/672
https://github.com/janhq/nitro/pull/155/
